kubernetes-itests/src/test/java/io/fabric8/kubernetes/PodIT.java

```diff
@@ -169,6 +169,39 @@ public void onClose(int i, String s) {
     assertNotNull(out.toString());
   }
 
+  @Test
+  public void execEchoCommand() throws InterruptedException {
+    // Wait for resources to get ready
+    ReadyEntity<Pod> podReady = new ReadyEntity<>(Pod.class, client, pod1.getMetadata().getName(), currentNamespace);
+    await().atMost(30, TimeUnit.SECONDS).until(podReady);
+    final CountDownLatch execLatch = new CountDownLatch(1);
+    ByteArrayOutputStream out = new ByteArrayOutputStream();
+    ExecWatch execWatch = client.pods().inNamespace(currentNamespace).withName(pod1.getMetadata().getName())
+      .writingOutput(out).withTTY().usingListener(new ExecListener() {
+        @Override
+        public void onOpen(Response response) {
+          logger.info("Shell was opened");
+        }
+
+        @Override
+        public void onFailure(Throwable throwable, Response response) {
+          logger.info("Shell barfed");
+          execLatch.countDown();
+        }
+
+        @Override
+        public void onClose(int i, String s) {
+          logger.info("Shell closed");
+          execLatch.countDown();
+        }
+      }).exec("echo 'H$ll* (W&RLD}'"); // These arguments in the exec command should be escaped.
```

Comment:
I think you need to separate the arguments. 
Also the `}` in `(W&RLD}` looks like a typo

Reply:
No I added that bracket just to demonstrated that all of these characters are being escaped properly.

Reply:
I understand, but then the assertion was inconsistent (changed now).